sno/working_copy/gpkg.py

```diff
@@ -89,57 +69,38 @@ def session(self, bulk=0):
         """
         L = logging.getLogger(f"{self.__class__.__qualname__}.session")
 
-        if hasattr(self, "_db"):
+        # TODO - look into bulk, locking_mode, journal_mode, synchronous, cache_size
+
+        if hasattr(self, "_session"):
             # inner - reuse
             L.debug(f"session(bulk={bulk}): existing...")
-            with self._db:
-                yield self._db
+            yield self._session
             L.debug(f"session(bulk={bulk}): existing/done")
+
         else:
             L.debug(f"session(bulk={bulk}): new...")
-            self._db = gpkg.db(
-                self.full_path,
-            )
-            dbcur = self._db.cursor()
 
-            if bulk:
-                L.debug("Invoking bulk mode %s", bulk)
-                orig_journal = dbcur.execute("PRAGMA journal_mode;").fetchone()[0]
-                orig_locking = dbcur.execute("PRAGMA locking_mode;").fetchone()[0]
-
-                dbcur.execute("PRAGMA synchronous = OFF;")
-                dbcur.execute("PRAGMA cache_size = -1048576;")  # -KiB => 1GiB
+            try:
+                self._session = self.sessionmaker()
 
+                if bulk:
+                    self._session.execute("PRAGMA synchronous = OFF;")
```

Comment:
see comments earlier, but we should avoid this stuff except for initial WC creation.

Reply:
@rcoup What counts as initial WC creation - if we need to drop and rewrite an entire table at some point, does that count?

Reply:
no, because the user might have other data in the DB that isn't committed or isn't Sno-managed at all. I think solely "we're creating a new GPKG file and populating it".

[SQLite docs](https://www.sqlite.org/howtocorrupt.html#_disabling_sync_using_pragmas):
> Setting `PRAGMA synchronous=OFF` can cause the database to go corrupt if there is an operating-system crash or power failure, though this setting is safe from damage due to application crashes.